### PR TITLE
Added mongoidX_or_newer? and mongoidX_or_older? helpers.

### PR DESCRIPTION
**Pull request by dblock**
@joeyAghion Care to review? It's a little tricky with betas because `7.0.0.beta` < `7.0`, but we don't want that.

Closes https://github.com/mongoid/mongoid-compatibility/issues/8

### CODE REVIEW

File: lib/mongoid/compatibility/version.rb [hunk 1]
```diff
@@ -8,6 +8,14 @@ module ClassMethods
...
             !!Mongoid::Compatibility::Version::ClassMethods.const_get("V#{v}")
           end
+
+          define_method "mongoid#{v}_or_newer?" do
+            Gem::Version.new(Mongoid::VERSION) >= Gem::Version.new("#{v - 1}.999")
+          end
+
+          define_method "mongoid#{v}_or_older?" do
+            Gem::Version.new(Mongoid::VERSION) < Gem::Version.new("#{v}.999")
+          end
```
Comment (joeyAghion) [inline lib/mongoid/compatibility/version.rb]: This strikes me as too custom. I didn't know `Gem::Version` existed, but now that I do, couldn't we just add a `gem_version` method that returns `Gem::Version.new(Mongoid::VERSION)`? We could then do straightforward comparisons like `.gem_version.segments.first >= 6`.

Comment (dblock) [reply lib/mongoid/compatibility/version.rb]: I fill uncomfortable comparing numbers, but comparing gem versions is the way to go I think, I changed it to `Gem::Version.new(Mongoid::VERSION).release <= Gem::Version.new(v.to_s)`, so the beta problem goes away.

Comment (dblock) [reply lib/mongoid/compatibility/version.rb]: .... scratch my last comment, `segments` should work better

**Comment by dblock** [comment]
@joeyAghion What do you think of this?

**Comment by joeyAghion** [comment]
Looks pretty straightforward. Now that I know about `Gem::Version` I _may_ just depend on that in the future, but if you're in favor of additions to the gem of this sort, I', 👍.

**Comment by yads** [comment]
Can we get a release with this feature enabled?

**Comment by joeyAghion** [comment]
Sure. See [0.5.1](https://rubygems.org/gems/mongoid-compatibility/versions/0.5.1).

**Pull request merged**